packages/api/utils/validation.ts

```diff
@@ -207,3 +223,14 @@ export function isFundingAmount(str: string): boolean {
   const num: number = parseFloat(str);
   return num > 0;
 }
+
+/**
+ * Check whether an operator email is a valid email.
+ *
+ * @param email - The email to check
+ * @returns boolean - true if the email matches the regex pattern, and false if not
+ */
+export function isMilEmail(email: string): boolean {
+  const emailRegex = /^[\w.-]+@[\w-]+.mil$/i;
```

Comment:
So... are we 100% that all `.mil` email addresses will match this regex? Office 365 is migrating towards allowing `+` extensions and Gmail already supports them. Not sure about Exchange (on-prem). So for example, we would reject something like `email-address+atat@domain.mil`. We actually use email addresses with `+` extensions _today_ internally for our ATAT group mailboxes.

This also has a pretty significant flaw after the `@` by not allowing subdomains. So unless you're directly under `.mil`, your address cannot be used. For example, `user.name@a.b.mil` would be rejected. But we can't just add `.` to the character class because then we'd accept things with multiple `.`s back-to-back.

I believe that with edge cases (such as quoted strings and comments), the language for email addresses may not even be regular; of course, we can accept tradeoffs, but I think forbidding `+` and subdomains may not be acceptable ones.

What if we did something a little more generic like:

```typescript
// All emails have to at least contain "@"
if (!email.contains("@")) {
  return false;
}

// Users should not input email addresses with comments
// so we can safely reject those. They aren't part of the actual
// address and so get dropped when the email is being sent anyway.
if (email.contains("(")) {
  return false;
}

// We make the assumption that we will not be receiving
// email addresses with quoted local parts so that we can
// ignore the complexities that come with parsing those.
if (email.contains("\"")) {
  return false;
}

// Split into the two halves
const [localPart, domain] = email.split("@");

// Reject values that are too long
if (localPart.length > 63 || domain.length > 253) {
  return false;
}

// Domain restrictions (per DNS spec and business logic)
// No need to handle IP address logic; those are rejected
// by business logic
// We're required to end in .mil per business logic and no
// domain can have any segment > 63 characters.
domainRegex = /^([a-z0-9-]{1,63}\.)+mil$/i;
if (!domainRegex.test(domain)) {
  return false;
}

// RFC 5321 defines the local part as a Dot-string (or Quoted-string, though we've
// rejected all those possible strings), which is an Atom followed by a dot and then
// an Atom any number of times.  An Atom is defined as one or more characters
// in the atext class, which per the errata is defined to be the same as in RFC 5322;
// the definition can be found at:
//   https://datatracker.ietf.org/doc/html/rfc5322#section-3.2.3.
// This lazily encodes that as a regular expression with some logic to forbid
// back-to-back `.` or starting/ending with `.`. Trying to do this in regex would likely
// require backtracking or other issues which might lead to a ReDoS vulnerability.
localPartCharacters = /^[a-z0-9!#$%&'*+\/=?^_`{|}~.-]+$/i
return (
  !localPart.startsWith(".") &&
  !localPart.endsWith(".") &&
  !localPart.contains("..") &&
  localPartCharacters.test(localPart)
);
```

Maybe this is a little too much correctness but with the comments stripped it's not really all that much longer than some other validation performed in the same file. I've been burned before about making assumption on what an email address looks like in an environment. Of course, we probably want to make sure that the front-end isn't performing more restrictive validation.

Regardless of whether the proposed start of a solution here is viable, we do want to make sure that we accept subdomains and `+`s.

Reply:
Nope, naive regex for sure. I didn't see this message until after I made another regex... 
```typescript
const emailRegex = /^[^.]([\w+-]+)(.)?([\w+-]+)?[^.]@([\w-]+.)+mil$/i
```
I initially started my approach similar to how you did, but decide to go with regex (less code). Although, like you pointed out my test cases missed a lot of edge cases. I've updated to include your approach.

Changed the strings to use `includes` since `contains` is not on strings. I left in most comments for others.

Reply:
Since the API spec is specifying format for email properties, and API Gateway will enforce the most common rules, can we make this minimal, such as ending in .mil?  Unless we need it otherwise.  But I think we're already prevented from receiving just any string by API Gateway.
```yaml
        email:
          type: string
          format: "email"
```